File: snippets/ruby/view_a_contact_s_events.rb
referral_code = "THE_REFERRAL_CODE"
list_id = "YOUR_LIST_ID"
user = fakeDB.get_user_by_referral_code(referral_code)
subscriber_hash = Digest::MD5.hexdigest(user.email.downcase)

response = mailchimp.lists.get_list_member_events(
  list_id,
  subscriber_hash
)

puts "We have created #{response["total_items"]} events for this user."

File: snippets/node/view_a_contact_s_events.js
const referralCode = "THE_REFERRAL_CODE";
const listId = "YOUR_LIST_ID";
const user = fakeDB.getUserByReferralCode(referralCode);
const subscriberHash = md5(user.email.toLowerCase());

async function run() {
  const response = await mailchimp.lists.getListMemberEvents(
    listId,
    subscriberHash
  );

  console.log(`We have created ${response.totalItems} events for this user.`);
}

run();